internal/pkg/template/upgrade_test.go

```diff
@@ -0,0 +1,155 @@
+			Value: 789,
+		},
+		{
+			Id:    "input-3",
+			Value: 3.50,
+		},
+		{
+			Id:    "input-4",
+			Value: false,
+		},
+	}
+
+	// Template
+	templateRef := model.NewTemplateRef(model.TemplateRepository{Name: "my-repository"}, "my-template", model.ZeroSemVersion())
+	instanceId := "my-instance"
+
+	// Current project state
+	d := dependenciesPkg.NewTestContainer()
+	projectState := d.EmptyState()
+	configKey := model.ConfigKey{BranchId: targetBranch.Id, ComponentId: "foo.bar", Id: "12345"}
+	rowKey := model.ConfigRowKey{BranchId: targetBranch.Id, ComponentId: "foo.bar", ConfigId: "12345", Id: "67890"}
+	configMetadata := make(model.ConfigMetadata)
+	configMetadata.SetTemplateInstance(templateRef.Repository().Name, templateRef.TemplateId(), instanceId)
+	configMetadata.SetConfigTemplateId("my-config")
+	configMetadata.AddRowTemplateId("67890", "my-row")
+	assert.NoError(t, projectState.Set(&model.ConfigState{
+		ConfigManifest: &model.ConfigManifest{ConfigKey: configKey},
+		Remote: &model.Config{
+			ConfigKey: configKey,
+			Metadata:  configMetadata,
+		},
+	}))
+	assert.NoError(t, projectState.Set(&model.ConfigRowState{
+		ConfigRowManifest: &model.ConfigRowManifest{ConfigRowKey: rowKey},
+		Remote: &model.ConfigRow{
+			ConfigRowKey: rowKey,
+		},
+	}))
+
+	// Create context
+	fs := testfs.NewMemoryFs()
+	ctx := NewUpgradeContext(context.Background(), templateRef, fs, instanceId, targetBranch, inputsValues, tickets, projectState)
+
+	// Check JsonNet functions
+	code := `
+{
+	Input1: Input("input-1"),
+    Input2: Input("input-2"),
+    Input3: Input("input-3"),
+    Input4: Input("input-4"),
+    Objects: {
+      Config1: ConfigId("my-config"),
+      Config2: ConfigId("my-config"),
+      Config3: ConfigId("new-config"),
+      Row1: ConfigRowId("my-row"),
+      Row2: ConfigRowId("my-row"),
+      Row3: ConfigRowId("new-row"),
+    },
+}
+`
+	expectedJson := `
+{
+  "Input1": "my-value-1",
+  "Input2": 789,
+  "Input3": 3.5,
+  "Input4": false,
+  "Objects": {
+    "Config1": "<<~~placeholder:1~~>>",
+    "Config2": "<<~~placeholder:1~~>>",
+    "Config3": "<<~~placeholder:3~~>>",
+    "Row1": "<<~~placeholder:2~~>>",
+    "Row2": "<<~~placeholder:2~~>>",
+    "Row3": "<<~~placeholder:4~~>>"
+  }
+}
+`
+	jsonOutput, err := jsonnet.Evaluate(code, ctx.JsonNetContext())
+	assert.NoError(t, err)
+	assert.Equal(t, strings.TrimLeft(expectedJson, "\n"), jsonOutput)
+
+	// Check tickets replacement
+	data := orderedmap.New()
+	json.MustDecodeString(jsonOutput, data)
+	replacements, err := ctx.Replacements()
+	assert.NoError(t, err)
+	modifiedData, err := replacements.Replace(data)
+	assert.NoError(t, err)
+	modifiedJson := json.MustEncodeString(modifiedData, true)
+
+	expectedJson = `
+{
+  "Input1": "my-value-1",
+  "Input2": 789,
+  "Input3": 3.5,
+  "Input4": false,
+  "Objects": {
+    "Config1": "12345",
+    "Config2": "12345",
+    "Config3": "1001",
+    "Row1": "67890",
+    "Row2": "67890",
+    "Row3": "1002"
+  }
+}
```

Comment:
Pri upgrade sablony sa musia zachovat IDs existujucich objektov, a nove IDs sa generuju iba pre nove objekty (resp. ak si nejaku konfiguraciu uzivatel vymazal rucne).

Je to prva cast, ... druha cast su default hodnoty vstupov, to bude v dalsich PRs, teraz je tu iba tento unit test.